### PR TITLE
Remove HTMLCollectionElementsRevIter

### PR DESCRIPTION
**Pull request by KiChjang**
It's almost a duplicate of HTMLCollectionElementsIter, aside from that its node_iter is a PrecedingNodeIter. We can simply use dynamic dispatch for node_iter instead of making it a concrete type.

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10094)

### CODE REVIEW

**Comment by asajeffrey** [comment]
Seems like there is a trade-off of doing a dynamic method dispatch vs some code complexity. I'm a bit worried that the dynamic method dispatch will block the iterator being inlined, so we'll get less optimized code as a result.

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/10094#-:-KDGj5JVTA6FN2jfZoNW:1727935830)*

File: components/script/dom/htmlcollection.rs [hunk 1]
```diff
@@ -232,7 +232,7 @@ impl HTMLCollection {
 
 // TODO: Make this generic, and avoid code duplication
```
Comment (KiChjang) [inline components/script/dom/htmlcollection.rs]: The eventual is actually to solve this TODO comment here. Using dynamic dispatch is a first step towards that direction.

**Comment by asajeffrey** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[components/script/dom/htmlcollection.rs, line 233 \[r1\]](https://reviewable.io:443/reviews/servo/servo/10094#-KDGQF42szf6W9Dgkm-Q-r1-233:-KDOYZG6nQBkMEJgbPjY:1840033713) ([raw file](https://github.com/servo/servo/blob/4e81fd36b0662fc7798c3cc778d28765e291cc88/components/script/dom/htmlcollection.rs#L233)):*
How difficult would it be to implement something like `HTMLCollectionElementsIter<I,F>` where `I: Iterator<Item = Root<Node>>` and `F: CollectionFilter`? If that's quite a bit of effort, then this makes sense as a half-way house, but if it's routine then we might as well do the generic version.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/10094)*

**Comment by KiChjang** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[components/script/dom/htmlcollection.rs, line 233 \[r1\]](https://reviewable.io:443/reviews/servo/servo/10094#-KDGQF42szf6W9Dgkm-Q-r1-233:-KDOqM9PDV_59_aRTMV7:-1463345151) ([raw file](https://github.com/servo/servo/blob/4e81fd36b0662fc7798c3cc778d28765e291cc88/components/script/dom/htmlcollection.rs#L233)):*
I still have a couple of details that I'll need to iron out, e.g. should there even be a CollectionFilter? Should the HTMLCollection store a Box<Iterator<Item = Root<Node>>? I'd like to make this change right now just so that I have a clearer concept of what is expected when trying to refactor this piece of code.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/10094)*

**Comment by asajeffrey** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[components/script/dom/htmlcollection.rs, line 233 \[r1\]](https://reviewable.io:443/reviews/servo/servo/10094#-KDGQF42szf6W9Dgkm-Q-r1-233:-KDOsKJMzdBUFR_wJcm8:-8436265) ([raw file](https://github.com/servo/servo/blob/4e81fd36b0662fc7798c3cc778d28765e291cc88/components/script/dom/htmlcollection.rs#L233)):*
OK, as long as we don't end up with this being the permanent code :)

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/10094)*

**Pull request merged**